drv/vsc7448/src/spi.rs

```diff
@@ -0,0 +1,276 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::VscError;
+use drv_spi_api::SpiDevice;
+use ringbuf::*;
+use vsc7448_pac::{types::RegisterAddress, Vsc7448};
+
+#[derive(Copy, Clone, PartialEq)]
+enum Trace {
+    None,
+    Read { addr: u32, value: u32 },
+    Write { addr: u32, value: u32 },
+}
+
+/// This indicates how many bytes we pad between (writing) the address bytes
+/// and (reading) data back, during SPI transactions to the VSC7448.  See
+/// section 5.5.2 for details.  1 padding byte should be good up to 6.5 MHz
+/// SPI clock.
+pub const SPI_PAD_BYTES: u8 = 1;
+
+// Flags to tune ringbuf output while developing
+const DEBUG_TRACE_SPI: u8 = 1 << 0;
+const DEBUG_MASK: u8 = 0;
+
+/// Writes the given value to the ringbuf if allowed by the global `DEBUG_MASK`
```

Comment:
You are gonna love #390 once I figure out the Humility side.

Reply:
Until then, I turned this on to run unconditionally; the masking was useful when it was part of a larger collection of trace options, but now this is in a standalone file, so it doesn't crowd out other info.